src/v/kafka/server/request_context.h

```diff
@@ -185,15 +185,28 @@ class request_context {
               r.data.throttle_time_ms, throttle_delay_ms());
         }
 
-        vlog(
-          klog.trace,
-          "[{}:{}] sending {}:{} for {}, response {}",
-          _conn->client_host(),
-          _conn->client_port(),
-          ResponseType::api_type::key,
-          ResponseType::api_type::name,
-          _header.client_id,
-          r);
+        if (klog.is_enabled(ss::log_level::trace)) {
```

Comment:
Is this the logic we want?

If trace is enabled we still log errors at trace?

Reply:
This is how I understood the requirements. Log everything at trace. If someone only wants to see errored responses without the fire hose that is TRACE, then they turn on debug logging for kafka.

Reply:
I think maybe a misunderstanding here:

```
if (success) {
vlog(trace)
} else {
vlog(debug)
}
```

I _thought_ that was what we discussed.

Reply:
The reasoning being that it's strange that the level level of error responses changes when you enable trace logs.

Reply:
done.